modules/creating_a_blueprint.md

```diff
@@ -0,0 +1,170 @@
+---
+layout: default
+title: Creating a blueprint
+permalink: /modules/creating-a-blueprint/
+redirect_from:
+  - /creating_a_blueprint.html
+  - /modules/creating_a_blueprint.html
+sitemap:
+    priority: 0.7
+    lastmod: 2015-12-05T18:40:00-00:00
+---
+
+# <i class="fa fa-cube"></i> Creating a blueprint
+
+A JHipster blueprint is a Yeoman generator that is [composed](http://yeoman.io/authoring/composability.html) from a specific JHipster sub-generator to extend the functionality of that sub-generator. The blueprint can override any defined getter of the sub generator and provide its own templates and functionality.
+
+JHipster blueprints are listed on the [JHipster marketplace]({{ site.url }}/modules/marketplace/) with the `jhipster-blueprint` label.
+
+This allows to create third-party blueprints that can override a specific part of JHipster, say for example only the client side templates.
+
+## Using a blueprint
+
+To use a blueprint, run the below command
+
+```
+jhipster --blueprint <blueprint name>
+```
+
+## Example
+
+The [JHipster Kotlin](https://github.com/jhipster/jhipster-kotlin) blueprint replaces most of the server side Java code with equivalent Kotlin code.
+
+It is our official blueprint that showcases how you can create your own blueprint.
+
+The [JHipster Sample Blueprint](https://github.com/hipster-labs/generator-jhipster-sample-blueprint) shows how a client sub generator can be overriden.
+
+Or, you can use the [JHipster blueprint generator](https://github.com/jhipster/generator-jhipster-blueprint) to help you to initialize your blueprint. Choose the sub-generators that you would like to override while answering the questions.
```

Comment:
It would be helpful to show the command to use this below so users don’t have to leave the page.